services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -209,32 +211,64 @@ private List<AbstractVeniceService> createServices() {
     // Create jvm metrics object
     jvmStats = new VeniceJVMStats(metricsRepository, "VeniceJVMStats");
 
+    if (serverConfig.isSystemSchemaInitializationAtStartTimeEnabled()) {
```

Comment:
What would happen if multiple servers send concurrent `addValueSchema` to the leader controller with the same value schema contents, looks like `VeniceHelixAdmin.addValueSchema` might have an issue dealing with it?

Reply:
It uses the pre-defined schema id to add value schema. API `addValueSchema` with value schema id is idempotent. For example, even if there are multiple calls to add value schema v15 at the same time, we will end up having only one znode /value-schema/15.